tests/Unit/ApparentHorizons/Test_SpherePackIterator.cpp

```diff
@@ -0,0 +1,45 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "ApparentHorizons/SpherePackIterator.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+SPECTRE_TEST_CASE("Unit.ApparentHorizons.SpherePackIterator",
+                  "[ApparentHorizons][Unit]") {
+  /// [spherepack_iterator_example]
+  const size_t l_max = 4;
+  const size_t m_max = 3;
+  SpherePackIterator iter(l_max, m_max);  // stride is 1.
+  // Allocate space for a SpherePack array
+  std::vector<double> array(iter.array_size());
+  // Set each array element equal to l+m
+  for (iter.reset(); iter; ++iter) {
+    array[iter()] = iter.l() + iter.m();
+  }
+  /// [spherepack_iterator_example]
+
+  SpherePackIterator s(3, 2, 5);
+  CHECK(s.l_max() == 3);
+  CHECK(s.m_max() == 2);
+  CHECK(s.n_th() == 4);
+  CHECK(s.n_ph() == 5);
+  const std::vector<size_t> test_l = {0, 1, 1, 2, 2, 2, 3, 3, 3, 1, 2, 2, 3, 3};
+  const std::vector<size_t> test_m = {0, 0, 1, 0, 1, 2, 0, 1, 2, 1, 1, 2, 1, 2};
+  const std::vector<size_t> test_index = {0,  15, 20, 30, 35,  40,  45,
+                                          50, 55, 80, 95, 100, 110, 115};
+  size_t i = 0;
+  for (; s; ++s, ++i) {
+    CHECK(s.l() == test_l[i]);
+    CHECK(s.m() == test_m[i]);
+    CHECK(s() == test_index[i]);
+  }
+
+  CHECK(s(2, 1) == 35);
+  CHECK(s(2, -1) == 95);
+
+  test_copy_semantics(s);
+  const auto s_copy = s;
+  test_move_semantics(std::move(s), s_copy, 4_st, 2_st, 3_st);
```

Comment:
test non-equality and equality and the stream operator (see get_output in TestHelpers)

Reply:
test coefficient_array()

Reply:
test reset